meteor-app/imports/api/assessments/server/publications.js

```diff
@@ -14,4 +15,8 @@ Meteor.publish('serviceItems.all', () => {
 
 Meteor.publish('assessments.all', () => {
   return Assessment.find({})
+})
+
+Meteor.publish('logger.all', () => {
```

Comment:
As the database grows, and you have lots of assessments, this will quickly become a problem, because it publishes every event in the log collection. This is the problem that the parts ordering has, because it is publishing 17,000 records.

Your use of the subscription should just be to find log events for a particular assessment, so rename the publication to be `logger.assessment`, and it has a single parameter, which is the assessment id. The find statement will now be `return Assessment.find({assessmentId: id})` Similarly the subscription needs to be modified for the change of name, and also to pass that id down to the server.

This mechanism is pretty efficient, as it will only get a small amount of data when looking at a service, rather than fetching _everything_.

It would be worth looking at the other publications, as they will suffer from the same problem. It is well worth spending the time to fix this, as you will prevent the performance problem from happening.